lib/utils/net-utils.js

```diff
@@ -4,29 +4,24 @@ const Promise = require('bluebird')
 const net = require('net')
 
 const NetUtils = {
-  isPortAvailable (port, listenAddress) {
+  findAndBindAvailablePort (port, listenAddress) {
     return new Promise((resolve, reject) => {
       const server = net.createServer()
 
-      server.unref()
-      server.on('error', (err) => {
-        server.close()
-        if (err.code === 'EADDRINUSE' || err.code === 'EACCES') {
-          resolve(false)
-        } else {
-          reject(err)
-        }
-      })
-
-      server.listen(port, listenAddress, () => {
-        server.close(() => resolve(true))
-      })
+      server
+        .on('error', (err) => {
+          server.close()
+          if (err.code === 'EADDRINUSE' || err.code === 'EACCES') {
+            server.listen(++port, listenAddress)
+          } else {
+            reject(err)
+          }
+        })
+        .on('listening', () => {
```

Comment:
Really nice :+1:

Reply:
Just to be sure - did you tested if https://github.com/karma-runner/karma/issues/3011 don't occurs ?

Reply:
Since there's no clear repro steps, I've tried to test it on a project with thousands tests running in parallel - I haven't seen any issues.